openedx/core/djangoapps/content/learning_sequences/api/tests/test_outlines.py

```diff
@@ -1716,6 +1718,216 @@ def test_processor_only(
             removed_usage_keys = processor.usage_keys_to_remove(full_outline)
             assert len(removed_usage_keys) == expected_values_dict[learner_to_verify.username]
 
+@ddt.ddt
+class CohortPartitionGroupsTestCase(OutlineProcessorTestCase):  # lint-amnesty, pylint: disable=missing-class-docstring
+    """Tests for cohort partitions outline processor that affects outlines"""
+
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+        cls.visibility = VisibilityData(
+            hide_from_toc=False,
+            visible_to_staff_only=False
+        )
+
+    def _add_course_mode(
+        self,
+        course_key,
+        mode_slug=CourseMode.VERIFIED,
+        mode_display_name='Verified Certificate'
+    ):
+        """
+        Add a course mode to the test course_key.
+        Args:
+            course_key
+            mode_slug (str): the slug of the mode to add
+            mode_display_name (str): the display name of the mode to add
+            upgrade_deadline_expired (bool): whether the upgrade deadline has passed
+        """
+        signals.post_save.disconnect(update_masters_access_course, sender=CourseMode)
+        try:
+            CourseMode.objects.create(
+                course_id=course_key,
+                mode_slug=mode_slug,
+                mode_display_name=mode_display_name,
+                min_price=50
+            )
+        finally:
+            signals.post_save.connect(update_masters_access_course, sender=CourseMode)
```

Comment:
Do we need to set a course mode to test this? I succeeded to test this using just audit mode.

Reply:
Indeed, well spotted! Corrected.